From ec8beb2fa279b8d7b1efbd9f5414895b17dfbe8b Mon Sep 17 00:00:00 2001
From: Katarina Behrens <bubli@bubli.org>
Date: Fri, 5 Sep 2014 14:11:15 +0200
Subject: [PATCH] fdo#82681: return IsUserDefined w/o negation

when the predicate is called StyleSheetIsUserDefinedPredicate

Returning negation thereof somewhere, somehow, corrupts one of
SfxItem pools. Thus, invalid read happens and crashes Impress
and Draw.

Now the loop in SdStyleSheetPool::UpdateStdNames does nothing
(it iterates through user-defined styles, but its body does
something for non-user-defined styles), so more investigation
here is needed.

This is partial revert of commit 3440766f0ad43454287a874b5fd34b6f2af6bdf1

Change-Id: I06f0da320c49674dac2adf43b7c374babbdadb57
(cherry picked from commit a784c33fd48eeb3394dc63610e384cf866233093)
Signed-off-by: Andras Timar <andras.timar@collabora.com>
---
 sd/source/core/stlpool.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sd/source/core/stlpool.cxx b/sd/source/core/stlpool.cxx
index 17ca4f0..c98dff7 100644
--- a/sd/source/core/stlpool.cxx
+++ b/sd/source/core/stlpool.cxx
@@ -943,7 +943,7 @@ struct StyleSheetIsUserDefinedPredicate : svl::StyleSheetPredicate
 
     bool Check(const SfxStyleSheetBase& sheet) SAL_OVERRIDE
     {
-        return !sheet.IsUserDefined();
+        return sheet.IsUserDefined();
     }
 };
 }
-- 
2.2.1

